src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -64,3 +66,236 @@ def test_WINNF_FT_S_SCS_5(self):
+    """
+    device_cert = os.path.join('certs', 'sas_ca_signed_client.cert')
+    device_key = os.path.join('certs', 'client.key')
+    self.assertTlsHandshakeFailure(device_cert, device_key)
+
+  @winnforum_testcase 
+  def test_WINNF_FT_S_SCS_11(self):
+    """Blacklisted Certificate presented during registration.
+    Checks that SAS UUT response with fatal alert message.
+    """
+    #Blacklist certificate
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    self._sas_admin.BlacklistByFccIdAndSerialNumber({'fccId': device_a['fccId'],'CbsdSerialNumber': device_a['cbsdSerialNumber']})
+    device_cert = os.path.join('certs', 'device_a.cert')
+    device_key = os.path.join('certs', 'device_a.key')
+    self.assertTlsHandshakeFailure(device_cert, device_key)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_12(self):
+    """Expired certificate presented during registration.
+    Checks that SAS UUT response with fatal alert message.
+    """
+    device_cert = os.path.join('certs', 'client_expired.cert')
+    device_key = os.path.join('certs', 'client_expired.key')
+    self.assertTlsHandshakeFailure(device_cert, device_key)
+
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_13(self):
+    """ Disallowed TLS method attempted during registration .
+        Checks that SAS UUT response with fatal alert message below
+        The SAS UUT sends a fatal alert message with the following parameters:
+         AlertLevel = 2 (fatal) --> tlsv1 alert protocol version
+         The SAS UUT immediately terminates the TLS session
+    """
+    device_cert = os.path.join('certs', 'client.cert')
+    device_key = os.path.join('certs', 'client.key')
+    self.assertTlsHandshakeFailure(device_cert, device_key,ciphers='CAMELLIA128-SHA',\
+                                     ssl_method=SSL.TLSv1_1_METHOD)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_14(self):
+    """Invalid ciphersuite presented during registration.
+    Checks that SAS UUT response with fatal alert message.
+    """
+    device_cert = os.path.join('certs', 'client.cert')
+    device_key = os.path.join('certs', 'client.key')
+    self.assertTlsHandshakeFailure(device_cert, device_key,ciphers='ECDHE-RSA-AES256-GCM-SHA384')
+
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_15(self):
+    """Certificate with inapplicable fields presented during registration. The response should be 104."""
+    
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+
+    device_a_cert = os.path.join('certs', 'client_inapplicable.cert')
+    device_a_key = os.path.join('certs', 'client.key')
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request,device_a_cert,device_a_key)['registrationResponse'][0]
```

Comment:
You have to specifically call     
  self.SasReset()
at the begin of this test to ensure that the SAS UUT is clean (as you are changing the SAS state).
See comment line 22

Reply:
Missed   the code change in last drop, added now